pkg/operation/botanist/secrets.go

```diff
@@ -441,6 +453,21 @@ func (b *Botanist) SyncShootCredentialsToGarden(ctx context.Context) error {
 		},
 	}
 
+	// ssh-keypair.old secret will be synced to the Garden cluster if it is present in shoot namespace in seed.
+	oldSecret := &corev1.Secret{}
+	err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.SecretNameOldSSHKeyPair), oldSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			b.Logger.Info("Newly created shoot will not have ssh-keypair.old secret")
+		}
+	} else {
+		projectSecrets = append(projectSecrets, projectSecret{
+			secretName: v1beta1constants.SecretNameOldSSHKeyPair,
+			suffix:     gutil.ShootProjectSecretSuffixOldSSHKeypair,
+			labels:     map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleSSHKeyPair},
+		})
+	}
```

Comment:
```suggestion
	if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.SecretNameOldSSHKeyPair), oldSecret); err != nil {
		if !apierrors.IsNotFound(err) {
			return err
		}
	} else {
		projectSecrets = append(projectSecrets, projectSecret{
			secretName: v1beta1constants.SecretNameOldSSHKeyPair,
			suffix:     gutil.ShootProjectSecretSuffixOldSSHKeypair,
			labels:     map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleSSHKeyPair},
		})
	}
```

Reply:
Addressed in [commit](https://github.com/gardener/gardener/pull/5388/commits/8e9970ad6576f23a3b8b79382aa9c8542ddd7ad4).